numpy/_core/tests/examples/float16/_float16_tests.c

```diff
@@ -0,0 +1,343 @@
+        return NULL;
+    }
+    int ret = npy_float16_lt_nonan(h1, h2);
+    return PyBool_FromLong(ret != 0);
+}
+
+static PyObject *
+float16tests_float16_le_nonan(PyObject *NPY_UNUSED(self), PyObject *args)
+{
+    unsigned int h1, h2;
+    if (!PyArg_ParseTuple(args, "II", &h1, &h2)) {
+        return NULL;
+    }
+    int ret = npy_float16_le_nonan(h1, h2);
+    return PyBool_FromLong(ret != 0);
+}
+
+/*
+ * Miscellaneous Functions
+ */
+static PyObject *
+float16tests_float16_iszero(PyObject *NPY_UNUSED(self), PyObject *args)
+{
+    unsigned int hbits;
+    if (!PyArg_ParseTuple(args, "I", &hbits)) {
+        return NULL;
+    }
+    int ret = npy_float16_iszero(hbits);
+    return PyBool_FromLong(ret != 0);
+}
+
+static PyObject *
+float16tests_float16_isnan(PyObject *NPY_UNUSED(self), PyObject *args)
+{
+    unsigned int hbits;
+    if (!PyArg_ParseTuple(args, "I", &hbits)) {
+        return NULL;
+    }
+    int ret = npy_float16_isnan(hbits);
+    return PyBool_FromLong(ret != 0);
+}
+
+static PyObject *
+float16tests_float16_isinf(PyObject *NPY_UNUSED(self), PyObject *args)
+{
+    unsigned int hbits;
+    if (!PyArg_ParseTuple(args, "I", &hbits)) {
+        return NULL;
+    }
+    int ret = npy_float16_isinf(hbits);
+    return PyBool_FromLong(ret != 0);
+}
+
+static PyObject *
+float16tests_float16_isfinite(PyObject *NPY_UNUSED(self), PyObject *args)
+{
+    unsigned int hbits;
+    if (!PyArg_ParseTuple(args, "I", &hbits)) {
+        return NULL;
+    }
+    int ret = npy_float16_isfinite(hbits);
+    return PyBool_FromLong(ret != 0);
+}
+
+static PyObject *
+float16tests_float16_signbit(PyObject *NPY_UNUSED(self), PyObject *args)
+{
+    unsigned int hbits;
+    if (!PyArg_ParseTuple(args, "I", &hbits)) {
+        return NULL;
+    }
+    int ret = npy_float16_signbit(hbits);
+    return PyBool_FromLong(ret != 0);
+}
+
+static PyObject *
+float16tests_float16_copysign(PyObject *NPY_UNUSED(self), PyObject *args)
+{
+    unsigned int xbits, ybits;
+    if (!PyArg_ParseTuple(args, "II", &xbits, &ybits)) {
+        return NULL;
+    }
+
+    npy_half x = xbits;
+    npy_half y = ybits;
+
+    npy_half r = npy_float16_copysign(x, y);
+    return PyLong_FromUnsignedLong(r);
+}
+
+static PyObject *
+float16tests_float16_nextafter(PyObject *NPY_UNUSED(self), PyObject *args)
+{
+    unsigned int xbits, ybits;
+    if (!PyArg_ParseTuple(args, "II", &xbits, &ybits)) {
+        return NULL;
+    }
+
+    npy_half x = xbits;
+    npy_half y = ybits;
+
+    npy_half r = npy_float16_nextafter(x, y);
+    return PyLong_FromUnsignedLong(r);
+}
+
+static PyObject *
+float16tests_float16_divmod(PyObject *NPY_UNUSED(self), PyObject *args)
+{
+    unsigned int h1bits, h2bits;
+    if (!PyArg_ParseTuple(args, "II", &h1bits, &h2bits)) {
+        return NULL;
+    }
+
+    npy_half h1 = h1bits;
+    npy_half h2 = h2bits;
+
+    npy_half mod;
+    npy_half quo = npy_float16_divmod(h1, h2, &mod);
+
+    return Py_BuildValue("II", quo, mod);
```

Comment:
This seems incorrect on some platforms. You need to make sure the types match. I assume `I` is `unsigned int`, so you have to type or cast `mod` and `quo` accordingly.  (`PyLong_FromUnsignedLong` is OK, since it is a typed function.)
(I have not checked if there are more occurrences of this)

The same is true for `ParseTuple`, but I think there it is correct everywhere.